File: 백준/Gold/1806. 부분합/부분합.py
import sys
input= sys.stdin.readline
n,s=map(int,input().split())
arr=list(map(int,input().split()))
l,r=0,0
sum=0
min_length=1e9
while True:
    if sum>=s:
        min_length=min(min_length,r-l)
        sum-=arr[l]
        l+=1
    elif r==n:
        break
    else:
        sum+=arr[r]
        r+=1
if min_length==1e9:
    print(0)
else:
    print(min_length)

File: 백준/Gold/1806. 부분합/부분합.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int s = scanner.nextInt();
        int[] arr = new int[n+1];
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }
        int l=0,r=0,sum=0,cnt=Integer.MAX_VALUE;
        while(l<=n&&r<=n){
            if(sum>=s){
                cnt=Math.min(cnt,r-l);
                sum-=arr[l++];
            }else{
                sum+=arr[r++];
            }
        }
        System.out.println(cnt==Integer.MAX_VALUE?0:cnt);
    }
}